katdal/chunkstore.py

```diff
@@ -477,62 +477,25 @@ def has_array(self, array_name, chunks, offset=()):
         success : :class:`numpy.ndarray` object
             Array of bools indicating presence of each chunk
 
-        Raises
-        ------
-        NotImplementedError
-            If the underlying store does not have an efficient implementation
+        Notes
+        -----
+        If the underlying store implements :meth:`list_chunk_ids`, that is
```

Comment:
@ludwigschwardt - why not just mandate that ALL stores implement list_chunk_ids (since the important ones already do). I see that the dict store has neither list_chunk_ids nor has_chunk...

Reply:
Dict store cannot support list_chunk_ids. It has no set chunks and allows arbitrary chunking via getitem. It does support has_chunk (look at the base class).

Reply:
Maybe that's something we should change, so that it works more like the others i.e. for each array it stores a dictionary mapping chunk ID to a numpy array?